manifests/server/config.pp

```diff
@@ -256,12 +258,13 @@
           content => template('postgresql/systemd-override.erb'),
           require => File['systemd-conf-dir'],
       ;
+    }
 
-      # Remove old unit file to avoid conflicts
-      'old-systemd-override':
-          ensure => absent,
-          path   => "/etc/systemd/system/${service_name}.service",
-      ;
+    # Remove old unit file to avoid conflicts
+    file {'old-systemd-override':
+      ensure => absent,
+      path   => "/etc/systemd/system/${service_name}.service",
+      before => $systemd_notify,
```

Comment:
Technically you could add before in the old style as well to keep the diff smaller. In fact, it already has a `notify => $systemd_notify` in the default so this looks redundant.

Reply:
perfect.Thank you